zipline/src/commonMain/kotlin/app/cash/zipline/internal/bridge/inbound.kt

```diff
@@ -59,29 +62,33 @@ internal class InboundCall(
   val funName: String,
   val encodedArguments: Array<String>,
 ) {
+  private val arguments = ArrayList<Any?>(encodedArguments.size / 2)
+  private var callStartResult: Any? = null
   private var i = 0
 
   fun <T> parameter(serializer: KSerializer<T>): T {
     while (i < encodedArguments.size) {
-      when (encodedArguments[i]) {
-        LABEL_VALUE -> {
-          val result = context.json.decodeFromStringFast(serializer, encodedArguments[i + 1])
-          i += 2
-          return result
-        }
-        LABEL_NULL -> {
-          i += 2
-          return null as T
-        }
+      val result = when (encodedArguments[i]) {
+        LABEL_VALUE -> context.json.decodeFromStringFast(serializer, encodedArguments[i + 1])
+        LABEL_NULL -> null as T
         else -> {
-          i += 2 // Ignore unknown argument.
+          // Ignore unknown argument.
+          i += 2
+          continue
         }
       }
+      arguments += result
+      i += 2
+      if (i == encodedArguments.size) {
+        callStartResult = context.endpoint.eventListener.callStart(context.name, context.service, funName, arguments)
+      }
```

Comment:
Not sure about this as a hook to trigger a side-effect. I would expect the function to which this instance is passed (once all parameters are added) to invoke the side-effect.

Reply:
Maybe this:

```
"fun ping(app.cash.zipline.SampleRequest): app.cash.zipline.SampleResponse" -> {
  val request = inboundCall.parameter(requestSerializer)
  inboundCall.callStart()
  val response = service.ping(request)
  inboundCall.callEnd(
    responseSerializer,
    response
  )
}
```

(I’m a bit reluctant to change generated code just ’cause everytime I do it takes a whole day.)